cypress/support/commands.js

```diff
@@ -23,3 +23,235 @@
+  const selectComboBoxInput = (selector, keyword) => {
+    cy.get('[data-test-subj="' + selector + '"]').find('[data-test-subj="comboBoxInput"]').trigger('focus').type('{selectall}{backspace}' + keyword)
+    cy.get('[data-test-subj="comboBoxOptionsList ' + selector + '-optionsList"]').find('[title="' + keyword + '"]').trigger('click', { force: true })
+  }
+
+  // Attempt up to three times to select the desired field and operator options and input the value (if applicable) 
+  const tryToAddFilter = (field, operator, value = null, retry = 0) => {
+    cy.wait(3000)
+    cy.get('[data-test-subj="addFilter"]').click({ scrollBehavior: 'center' }).then(() => {
+      
+      selectComboBoxInput('filterFieldSuggestionList', field)
+      cy.get('[data-test-subj="filterFieldSuggestionList"]').then(($field) => {
+        const cls = $field.attr('class')
+        if (cls.includes('euiComboBox-isInvalid') && retry < 3) {
+          cy.get('[data-test-subj="cancelSaveFilter"]').click()
+          tryToAddFilter(field, operator, value, retry + 1)
+        }
+        else {
+          selectComboBoxInput('filterOperatorList', operator)
+          cy.get('[data-test-subj="filterOperatorList"]').then(($operator) => {
+            const cls = $operator.attr('class')
+            if (cls.includes('euiComboBox-isInvalid') && retry < 3) {
+              cy.get('[data-test-subj="cancelSaveFilter"]').click()
+              tryToAddFilter(field, operator, value, retry + 1)
+            }
+            else {
+              if(value !== null) {
+                cy.get('[data-test-subj="filterParams"]').find('input').type(value)
+              }
+              cy.get('[data-test-subj="saveFilter"]').click()
+            }
+          })
+        }
+      })
+    })
+  }
+  tryToAddFilter(field, operator, value)
+})
+
+/**
+ * Attempts to add a specified filter to a dashboard, retrying by reselecting the failed option
+ * @param {String} field Field value to select
+ * @param {String} operator Operator value to select
+ * @param {String} value Value field input
+ */
+
+Cypress.Commands.add('addDashboardFilterRetrySelection', (field, operator, value = null) => {
+  const selectComboBoxInput = (selector, keyword, retry = 0) => {
+    cy.get('[data-test-subj="' + selector + '"]').find('[data-test-subj="comboBoxInput"]').trigger('focus').type('{selectall}{backspace}' + keyword)
+    cy.get('[data-test-subj="comboBoxOptionsList ' + selector + '-optionsList"]').find('[title="' + keyword + '"]').trigger('click', { force: true })
+    cy.get('[data-test-subj="' + selector + '"]').then(($box) => {
+      const cls = $box.attr('class')
+      if (cls.includes('euiComboBox-isInvalid') && retry < 3) {
+        cy.wrap($box).find('[data-test-subj="comboBoxInput"]').type('{selectall}{backspace}')
+        cy.wrap($box).find('[data-test-subj="comboBoxToggleListButton"]').click()
+        selectComboBoxInput(selector, keyword, retry + 1)
+      }
+    })
+  }
```

Comment:
Do you think it's better to have retries built into each function or have a retry wrapper when calling the functions?

Reply:
I think it would be difficult to make a general-purpose retry wrapper since the condition that would determine whether a function should be retried would be very different between UI elements, and the method to "reset" the state of the UI element in preparation for a reset would also be difficult to generalize. However, I agree that having a wrapper retry function would make the code easier to maintain, and the feasibility of such a wrapper should further explored.